### PR TITLE
Web6169 handle put book errors

### PR DESCRIPTION
**Pull request by kevinhodges**
#### What are the links to relevant tickets?
https://hxshortbreaks.atlassian.net/browse/WEB-6169

#### What does this PR do?
Allows The Works to know when a Revolver put::book fails

#### What functional/unit tests does this PR have?
I've covered the additional functionality  in unit tests.

#### How should a developer review this?
Bump the version of `plugin-revolver` in The Works from `0.2.0` to `0.3.0` and check that when a firm fails the controller knows why and that when it's successful, we can operate as normal.

#### Any background context you want to provide?
None

#### Screenshots (if appropriate)
---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [x] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [x] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.

#### Reviewer 1 @philmeehan
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

#### Reviewer 2 @t1gr0u
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

File: lib/pluginRevolver.js [hunk 1]
```diff
@@ -113,6 +87,69 @@ function pluginRevolver( server, options, next ) {
...
+			if( isSuccess ) {
+				// check if this one was successful
+				isSuccess = ( element.success !== false && element.success !== 'false' );
+			}
+		} );
+		return isSuccess;
+	}
+
+
+	/**
```
Comment (philmeehan) [inline lib/pluginRevolver.js]: missing comment / or comment syntax created by accident

File: lib/pluginRevolver.js [hunk 2]
```diff
@@ -113,6 +87,69 @@ function pluginRevolver( server, options, next ) {
 
 	}
 
+	function _hasPutBookSucceeded( firmedElements ) {
```
Comment (philmeehan) [inline lib/pluginRevolver.js]: This passed your unit tests
```
	function _hasPutBookSucceeded( firmedElements ) {
		return _.reduce( firmedElements, function( isSuccess, element ) {
			// returns false if previous element didn't firm or this one had a success = false (checking both boolean and string)
			return isSuccess && ( element.success !== false && element.success !== 'false' );
		}, true );
	}
```

Comment (kevinhodges) [inline lib/pluginRevolver.js]: +1, thanks for doing this, I tried, I REALLY TRIED!

Comment (philmeehan) [inline lib/pluginRevolver.js]: It was based on your forEach so you were really close.

File: test/pluginRevolverTest.js [hunk 3]
```diff
@@ -121,211 +121,226 @@ describe( 'pluginRevolver', function() {
...
+					sinon.stub( request, 'del' ).yields( null, null, loadTestResource( './fixtures/fakeResponse' ) );
+				} );
 
-			it( 'should handle an http error from the stubbed server', function() {
-				return plugin.makeItSo( loadTestResource( './fixtures/fakeRequest' ) ).then( function() {
-				}, function( error ) {
-					error.should.have.property( 'error' ).that.is.an.instanceof( Error );
-					error.error.message.should.equal( 'requestFailed' );
+				after( function() {
+					request['del'].restore();
```
Comment (t1gr0u) [inline test/pluginRevolverTest.js]: `request.del.restore()`

File: test/pluginRevolverTest.js [hunk 4]
```diff
@@ -121,211 +121,226 @@ describe( 'pluginRevolver', function() {
...
-					error.error.message.should.equal( 'server error' );
+				before( function() {
+					sinon.stub( request, 'del' ).yields( 'fail' );
 				} );
-			} );
-		} );
 
-		describe( 'request', function() {
+				after( function() {
+					request['del'].restore();
```
Comment (t1gr0u) [inline test/pluginRevolverTest.js]: `request.del.restore();`

File: test/pluginRevolverTest.js [hunk 5]
```diff
@@ -121,211 +121,226 @@ describe( 'pluginRevolver', function() {
...
 			} );
+		} );
 
-			after( function( done ) {
-				request.get.restore();
-				done();
+		describe( 'when one of two elements is false (string)', function() {
+			it( 'should return false', function() {
+				var firmedElements = [ {
+					reference: '12345',
```
Comment (t1gr0u) [inline test/pluginRevolverTest.js]: we would normally do `reference: reference1`, unless we have some logic which check the value.

**Pull request merged**